### PR TITLE
SPORTSHUB-263 Pass Through Stripe Fees to Customer

### PR DESCRIPTION
**Pull request by owya490**
## Why was your change needed?

Please detail here why you did the change you did.

## What was Changed?

Please detail here what were the main changes you made.

## Any extra notes for reviewers to know?

Write any extra notes you may have for the PR.
This may include specific code paths that need extra scrutiny, or how to review a specific portion of code.

### CODE REVIEW

File: frontend/components/organiser/dashboard/OrganiserAnnouncementBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <div className="flex flex-wrap items-center gap-x-4 gap-y-2">
+          <p className="text-sm leading-6 text-black">
+            <strong className="font-semibold">NEW FEATURE</strong>
+            <svg viewBox="0 0 2 2" aria-hidden="true" className="mx-2 inline h-0.5 w-0.5 fill-current">
+              <circle r={1} cx={1} cy={1} />
+            </svg>
+            Pass through Application Card Surcharges to customers.
+          </p>
+          <a
+            href="/event/create"
```
Comment (Edwin-Ni) [inline frontend/components/organiser/dashboard/OrganiserAnnouncementBanner.tsx:18]: It shows a "Please login/create an account" page as it loads create event page

File: frontend/interfaces/EventTypes.ts [hunk 2]
```diff
@@ -36,6 +36,7 @@ interface AbstractEventData {
   accessCount: number;
   sport: string;
   paymentsActive: boolean;
+  stripeFeeToCustomer?: boolean;
```
Comment (Edwin-Ni) [inline frontend/interfaces/EventTypes.ts:39]: This field should be compulsory right?

File: functions/lib/stripe/checkout.py [hunk 3]
```diff
@@ -37,6 +39,11 @@ def __post_init__(self):
     if not isinstance(self.successUrl, str):
       raise ValueError("Success Url must be provided as a string.")
 
+def calculate_stripe_fee(price: float):
+  # Stripe fee is 30c + 1.7% of total price as price passed in is in cents already
+  # can just do the calculation and return a whole integer
+  return int(30 + (price * 0.017))
```
Comment (41dan) [inline functions/lib/stripe/checkout.py:45]: round up?

**Review by Edwin-Ni — CHANGES_REQUESTED** [review]
Looks good, have a look at one comment

**Review by 41dan — APPROVED** [review]
lgtm

File: functions/lib/stripe/checkout.py (hunk 3 above)
Comment (owya490) [reply functions/lib/stripe/checkout.py]: done, made it a int(math.ceil(...))

**Pull request merged**